the-collab-lab/tcl-71-smart-shopping-list
# Apply styles to the List view

## Description
The list view is styled according to the [Figma](https://www.figma.com/file/zPeqMaa7CZs1lewYuNWat6/tcl-71-smart-shopping-list?type=design&node-id=0-1&mode=design&t=ropZExMhIYBX4Y0c-0) specifications.

The changes applied in this PR include the styling of two input elements, which presented the main challenge for this work. Styling the checkbox input was particularly tricky because to follow the design specs it was necessary to completely change its appearance, while ensuring the checkbox remains fully accessible and interactive.

The chosen approach was inspired by the blog post [Inclusively Hiding & Styling Checkboxes and Radio Buttons](https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/) and involved visually hiding the checkbox while keeping it on the screen and replacing its visual representation with an SVG image that matches our design requirements. Then, leverage the [Tailwind `peer class`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state) to apply the different needed styles to the SVG, based on the checkbox status.

### Note
Changes in this PR include setting the total width for the list view in the List.jsx component. This is something that we might want to change when we work on the Layout.jsx styles and set a common width from this component that applies to all the views.

## Related Issue

Closes #47 

## Acceptance Criteria

- [x] Fonts and colors are respected.
- [x] The icons have the correct color.
- [x] When being checked, the checkbox is being filled with color and the item text is strikethrough.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :art:  Enhancement |

## Updates

### Before
#### List without items
<img width="430" alt="Screenshot 2024-03-30 at 16 08 45" src="https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/91918142/572d16db-be05-4c08-a7ab-1b3e23243c93">


#### List with items
<img width="430" alt="Screenshot 2024-03-30 at 16 09 00" src="https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/91918142/c9596851-e7fe-4a65-a0a0-62139f28fb6a">

### After
#### List without items
<img width="430" alt="Screenshot 2024-03-30 at 16 08 25" src="https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/91918142/c9ad0980-e319-4d27-883e-a1d05ccf643b">

#### List with items
<img width="428" alt="Screenshot 2024-03-30 at 17 28 56" src="https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/91918142/9c21ac72-5695-49d3-8d6f-7930d74a49d2">



## Testing Steps / QA Criteria
- Checkout your local `styling` branch and make sure it is updated
`git checkout styling`
`git pull origin styling`
- Create and checkout the branch locally and pull the changes
`git checkout -b vi-list-view`
`git pull origin vi-list-view`
- See the changes applied in the UI
